src/dropdown/dropdown.ts

```diff
@@ -242,4 +253,47 @@ export class NgbDropdown implements OnInit, OnDestroy {
       this._menu.position(this._anchor.anchorEl, this.placement);
     }
   }
+
+  private getActiveElmIndex(list): number {
+    let i = 0;
+    let position = -1;
+    for (let elm of list) {
+      if (elm === document.activeElement) {
+        position = i;
+      }
+      i++;
+    }
+    return position;
+  }
+
+  keyboardEvent($event): boolean {
+    if (!this.isOpen()) {
+      this.open();
+    }
+    let list = this.getMenuElements();
+    let position: number = this.getActiveElmIndex(list);
+    switch ($event.which) {
+      case Key.ArrowDown:
+        position++;
+        break;
+      case Key.ArrowUp:
+        position--;
+        break;
+    }
+    if (position >= list.length) {
+      position = list.length - 1;
+    }
+    if (position < 0) {
+      position = 0;
+    }
+    let elm: HTMLElement = list[position];
+    if (elm) {
+      elm.focus();
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  private getMenuElements(): HTMLElement[] { return <HTMLElement[]>Array.from(this._menuRef.nativeElement.children); }
```

Comment:
I think it should be something like:

```ts
this.menuElementRef.nativeElement.quertSelectorAll(`.dropdown-item:not(.disabled):not(:disabled)`);
```

because you never know what's inside